src/components/search/dataSearchQueryBuilder.js

```diff
@@ -0,0 +1,61 @@
+export const getDataSimpleSearchQuery = (
+    searchTerm,
+    userHomeDir,
+    rowsPerPage,
+    offset,
+    sortField,
+    sortDir,
+) => {
+    if (userHomeDir) {
+        return {
+            query: {
+                all: [
+                    {
+                        type: "label",
+                        args: {
+                            exact: false,
+                            label: searchTerm,
+                        },
+                    },
+                    {
+                        type: "path",
+                        args: {
+                            prefix: userHomeDir,
+                        },
+                    },
+                ],
+            },
+            size: rowsPerPage,
+            from: offset,
+            sort: [
+                {
+                    field: sortField,
+                    order: sortDir,
+                },
+            ],
+        };
+    } else {
+        return {
+            query: {
+                all: [
+                    {
+                        type: "label",
+                        args: {
+                            exact: false,
+                            negated: false,
```

Comment:
It seems like the only difference between these objects is an extra "path" object in the `all` array? Except this "label" object has a `negated` arg, but I thought that wasn't needed anymore?

If they should be the same, then it might be easier to see this if this query in the `else` section was refactored out if the conditional, and into something like `const searchQuery = { query: ...}`, and then `if (userHomeDir) searchQuery.query.all.push({type: "path" ...})`, and finally `return searchQuery`.

Reply:
Yeah, `negated` can go away here too. Just since I like thinking about different ways of organizing code, here's my thought:

```javascript
const searchClauses = [ { type: "label", args: { exact: false, label: searchTerm } } ]
return {
  query: {
    all: (userHomeDir ?
      searchClauses.concat({ type: "path", args: { prefix: userHomeDir } }) : 
      searchClauses),
  },
  size: rowsPerPage,
  from: offset,
  sort: [ { field: sortField, order: sortDir } ]
}
```

I don't know whether it's clearer inlining the if with a ternary operator or doing it your way, though.